File: Completed/JavaScript/942.DIStringMatch.js
/**
 * @param {string} S
 * @return {number[]}
 */
var diStringMatch = function(S) {
  let low = 0, high = S.length
  let result = []
  S.split('').forEach((s) => {
      if (s === 'I') {
          result.push(low)
          low++
      } else {
          result.push(high)
          high--
      }
  })
  result.push(low)
  return result
};

File: Completed/Go/942.DIStringMatch.go
func diStringMatch(S string) []int {
	var low, high int = 0, len(S)
	result := []int{}
	
	for _, val := range S {
			if val == 'I' {
					result = append(result, low)
					low++
			} else {
					result = append(result, high)
					high--
			}
	}
	
	result = append(result, low)
	
	return result
}